### DISCUSSION TITLE
Capture additional values in the apply service

### DISCUSSION THREAD

**Discussion by ewild** [bullhorn/career-portal#394] [Q&A]
I have forked this project, and I am trying to add an additional field, specifically, the hourlyRate which we use to store the desired rate, to the 'apply' method. The Url for the POST looks like this:

https://public-restXX.bullhornstaffing.com/rest-services/XXXXX/apply/6789/raw?externalID=Resume&type=Resume&firstName=John&lastName=Test&email=JohnTest%40yahoo.com&phone=8005551212&format=pdf&hourlyRate=12.34

It saves everything other than the hourlyRate.

Is the public-rest service limited to the fields in the config along with the eeoc fields?

If so the alternative would be to create a custom API to then call the restXX.bullhornstaffing.com or see if Bullhorn has an interest in capturing this additional field in the public service?

-Eric

**Comment by charlesabarnes** [comment]
> Is the public-rest service limited to the fields in the config along with the eeoc fields?


Correct


> If so the alternative would be to create a custom API to then call the restXX.bullhornstaffing.com or see if Bullhorn has an interest in capturing this additional field in the public service?


You are right about these follow up options as well.

**Reply by ewild** [reply]
Thank you for your help. 
Do you know if the API is open source or is it maintained by Bullhorn Co?
Do you think there would be any community benefit of adding 'hourlyRate'? I don't know how that field is normally used in the application. In the app I am working with it is used as the desired pay rate.

**Reply by pbearne** [reply]
The public API is maintained by Bullhorn and is just enough to support the OSCP
The private API (also maintained by Bullhorn) has access to all of the objects in Bullhorn
We use the private API for Matador Jobs as the Public API is too limiting for the needs of our customers

**Reply by ewild** [reply]
Thank you for the help. 
This does seem like the way to go now. I have experience working with the private API so  I think I just need to duplicate what the Public does and add the additional field along the way.

**Reply by pbearne** [reply]
That is a lot of work 
If you are in WordPress land its all done in the Matador Plugin with just a couple of clicks in the settings
If you need the UI to be in react we have clients that let matador do the heavy lifting and work of the WP rest API

**Reply by ewild** [reply]
I am new to this project, so. We have a WordPress app that uses this career-portal to allow candidates to submit resumes for jobs to our Bullhorn instance. Are you saying that there are other templates for us to use to capture this info? Another Wordpress site or a React project?

**Reply by pbearne** [reply]
Hi Eric,

Matador Jobs is a Bullhorn partner that provides an alternative solution/job board for WordPress sites
Reach out to me at paul@matadorjobs.com and I give you a quick demo as I believe I can save a lot of time etc.
Paul

**Comment by inTulsa** [comment]
Hi @charlesabarnes - I'm wondering if you can expand here on the available options.

Is there any way to get a list of fields that are only available to us using the "public-restXX.bullhornstaffing"? I'm trying to capture data in the "referredBy" field along with the source, and like the OP describes, it saves everything but the referredBy value.

Could you also elaborate on whether there is a way to get Bullhorn to add an additional field for us to have access to write to? Sort of like exposing the field in the back-end somehow.

Thank you for your help.

**Reply by charlesabarnes** [reply]
The only fields available are the name fields, email, phone number, resume, source, and the EEOC fields.   The public rest endpoint was designed to be limited for security reasons since it is not behind any authentication.   Work to add additional fields to the public-rest endpoint is not currently a priority, the best avenue I can suggest is submitting a product feature request through https://help.bullhorn.com/s/feature-requests to add additional fields.     

Until then, I think the alternatives would be the ones mentioned.
